portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteApps.testcase

```diff
@@ -517,6 +514,56 @@ definition {
 		}
 	}
 
+	@description = "LPS-159013 ES module custom element portlets render incorrect <script> node when dragged in the UI"
+	@priority = "4"
+	test CustomElementRendersCorrectlyWhenPlacedOnWidgetPage {
+		property portal.acceptance = "true";
+		property testray.component.names = "Remote Apps";
+
+		task ("Given a custom element with "Use ES Modules" option created") {
+			JSONRemoteApp.addCustomElementRemoteAppEntry(
+				customElementCssurl = "https://liferay.github.io/liferay-frontend-projects/index.css",
+				customElementHtmlName = "vanilla-counter",
+				customElementName = "Vanilla Counter",
+				customElementProperties = "test-data-user=QAuser",
+				customElementURL = "https://liferay.github.io/liferay-frontend-projects/vanilla-counter/index.js",
+				customElementUseESM = "true");
+		}
+
+		task ("And Given a widget page created") {
+			JSONLayout.addPublicLayout(
+				groupName = "Guest",
+				layoutName = "My Widget Page");
+		}
+
+		task ("When adding the app portlet to the page") {
+			Navigator.gotoPage(pageName = "My Widget Page");
+
+			Refresh();
+
+			Click(locator1 = "ControlMenu#ADD");
+
+			Type(
+				locator1 = "NavBar#APPLICATION_SEARCH_FIELD",
+				value1 = "Vanilla Counter");
+
+			Portlet._clickAddPortlet(portletName = "Vanilla Counter");
+
+			WaitForElementPresent(locator1 = "Message#SUCCESS");
+		}
+
+		task ("Then the custom element renders correctly") {
+			AssertElementPresent(locator1 = "//vanilla-counter[@test-data-user='QAuser']");
+
+			Refresh();
+
+			AssertAttributeValue(
+				attribute1 = "src",
+				locator1 = "RemoteAppsEntry#CUSTOM_ELEMENT_MODULE_SCRIPT",
+				value1 = "https://liferay.github.io/liferay-frontend-projects/vanilla-counter/index.js");
+		}
```

Comment:
After many tries of locating the script of type module for Vanilla Counter without refreshing the page, we will first be checking the property of the portlet and then after refreshing the correct URL of the portlet. The URL before refreshing is not possible to locate, but it is available in sources (it happens in Chrome and FIrefox, hard to say what is the reason, cause the portlet renders correctly)

Reply:
:thinking: